waspc/src/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -0,0 +1,74 @@
+module Generator.WebAppGenerator.RouterGenerator
+    ( generateRouter
+    ) where
+
+import Data.Aeson (ToJSON(..), (.=), object)
+import qualified Path as P
+
+import StrongPath ((</>))
+import qualified StrongPath as SP
+
+import Wasp (Wasp)
+import qualified Wasp
+import qualified Wasp.Route
+import qualified Wasp.Page
+import qualified Wasp.JsImport
+
+import Generator.FileDraft (FileDraft)
+import Generator.WebAppGenerator.Common (asTmplFile, asWebAppSrcFile)
+import qualified Generator.WebAppGenerator.Common as C
+
+data RouterTemplateData = RouterTemplateData
+    { _routes :: ![Wasp.Route.Route]
+    , _pagesToImport :: ![PageTemplateData]
+    }
+
+instance ToJSON RouterTemplateData where
+    toJSON routerTD = object
+        [ "routes" .= _routes routerTD
+        , "pagesToImport" .= _pagesToImport routerTD
+        ]
+
+data PageTemplateData = PageTemplateData
+    { _importWhat :: !String
+    , _importFrom :: !String
+    } deriving (Show, Eq)
+
+instance ToJSON PageTemplateData where
+    toJSON pageTD = object
+        [ "importWhat" .= _importWhat pageTD
+        , "importFrom" .= _importFrom pageTD
+        ]
+
+generateRouter :: Wasp -> FileDraft
+generateRouter wasp = C.makeTemplateFD
+    (asTmplFile $ [P.reldir|src|] P.</> routerPath)
+    targetPath
+    (Just $ toJSON templateData)
+    where
+        routerPath = [P.relfile|router.js|]
+        templateData = createRouterTemplateData wasp
+        targetPath = C.webAppSrcDirInWebAppRootDir </> (asWebAppSrcFile routerPath)
+
+createRouterTemplateData :: Wasp -> RouterTemplateData
+createRouterTemplateData wasp = RouterTemplateData
+    { _routes = Wasp.getRoutes wasp
+    , _pagesToImport = map createPageTemplateData $ Wasp.getPages wasp
+    }
+
+createPageTemplateData :: Wasp.Page.Page -> PageTemplateData
+createPageTemplateData page = PageTemplateData
+    { _importFrom = relPathToExtSrcDir ++ SP.toFilePath  (Wasp.JsImport._from pageComponent)
+    , _importWhat = case Wasp.JsImport._namedImports pageComponent of
```

Comment:
You might want to check how I did smth similar here, just as an alternative idea: https://github.com/wasp-lang/wasp/pull/27/files#diff-8bc93b5297e70f9c095a8301f40646b1R43-R47 , but probably how you did it is fine, I am not sure it is even the same situation.

So what happens if they use default import? Identifier they use is actually ignored and we use our own identifie, pageName?

Reply:
yes, exactly that. We always end up with pageName.